python/truelayer_signing/verify.py

```diff
@@ -0,0 +1,227 @@
+from __future__ import annotations
+
+# std imports
+import json
+from collections import OrderedDict
+from dataclasses import dataclass
+from enum import Enum
+from typing import Dict, Iterable, List, Mapping, Optional, Tuple, Union
+from jwt import InvalidKeyError
+from json import JSONDecodeError
+
+# third party imports
+from jwt.algorithms import ECAlgorithm
+
+# local imports
+from .utils import (
+    HttpMethod,
+    TlJwsBase,
+    build_v2_jws_b64,
+    decode_url_safe_base64,
+    to_url_safe_base64
+)
+from .errors import TlSigningException
+
+
+class KeyFmt(Enum):
+    PEM = 0
+    JWKS = 1
+
+
+class TlVerifier(TlJwsBase[Union[str, Mapping[str, str]]]):
+    """
+    Tl-Verifier
+    """
+    key_fmt: KeyFmt
+    required_headers: List[str]
+
+    def __init__(
+        self,
+        pkey: Union[str, Mapping[str, str]],
+        key_fmt: KeyFmt,
+        method: HttpMethod = HttpMethod.POST,
```

Comment:
We don't default this in the verifier in other languages, so maybe best to be consistent & require it be specified here too.